### PR TITLE
[camera] Fix event type check

### PR DESCRIPTION
**Pull request by Zazo032**
## Description

Trivial change to properly check if the type of an event is an error, so a description can be added.

## Related Issues

Fixes flutter/flutter#41601

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
  - Trivial change
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java [hunk 1]
```diff
@@ -496,7 +496,7 @@ private void sendEvent(EventType eventType, String description) {
       Map<String, String> event = new HashMap<>();
       event.put("eventType", eventType.toString().toLowerCase());
       // Only errors have description
-      if (eventType != EventType.ERROR) {
+      if (eventType == EventType.ERROR) {
```
Comment (mklim) [inline packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java]: Whoops. Thank you for fixing this.

File: packages/camera/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.5.4+4
```
Comment (mklim) [inline packages/camera/CHANGELOG.md]: nit: This should be updated to `0.5.5+1` now.

**Comment by mklim** [comment]
Thank you for the fix!

We've recently re-emphasized that all PRs need tests, [following the style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#write-test-find-bug).

This is difficult for plugins PRs since we don't have an integrated testing solution, but I don't think it's impossible to do here. Normally I would recommend writing a flutter driver test that tries to do something invalid and then make sure the error has a description, but tests for this plugin need to somehow automatically accept the camera permission to run and we don't have a good cross platform solution for that yet.

We do have the ability to write JUnit tests though, and I think that would also work. Would you be able to create a CameraTests.java class with a unit test making sure messages are sent with a description? I think the easiest way to do this would be to open `example/app/android` in Android studio, and then use it to automatically make and run a new unit test class for `Camera.java`.

If you're not able to carry this forward, please let us know and we'll be happy to shepherd it on from here. Thanks again for the contribution.

**Comment by Zazo032** [comment]
I have been trying to generate a test class for `Camera.java` but Android Studio is ignorning me.

image

Pressing "OK" does nothing, tried checking/unchecking different methods, using Groovy/JUnit 4/JUnit 5 but Android Studio just ignores me, not sure if I'm missing something else.

**Comment by mklim** [comment]
Thanks for trying! I opened up Android Studio myself to add in a "quick test" skeleton for you to pick up and then realized that actually properly unit testing this fix is actually _really_ difficult. The class wasn't set up to be testable. I didn't see a way to do it without refactoring the `send()` logic out into a new class. Even instantiating `Camera` depends on having hardware mocked out at a level that was really difficult to do. I ended up at a point where I was trying to make custom Robolectric shadows for the camera hardware and still couldn't cover everything.

So instead I ended up making [two commits](https://github.com/mklim/plugins/compare/4c082f4...489867e), first to refactor the class and then to actually test this method. What do you think about cherry picking them into this PR and then we can land it all together?

**Comment by Zazo032** [comment]
Not sure if cherry picked or just merged (a bit new to this tools, excuse me if this isn't the correct process), but tests are passing correctly so this is probably ready to merge (waiting for review approvals)

**Pull request merged**